test/error/metal_threads_too_large.cpp

```diff
@@ -0,0 +1,33 @@
+#include "Halide.h"
+#include "halide_test_dirs.h"
+
+using namespace Halide;
+
+int main(int argc, char **argv) {
+    ImageParam im(UInt(16), 2, "input");
+    Func f("f");
+    Var x("x"), y("y");
+
+    f(x, y) = im(x, y) + 42;
+    f.gpu_blocks(y).gpu_threads(x, DeviceAPI::Metal);
```

Comment:
What happens if the machine doesn't support Metal?  I think this needs a check for whether the JIT target supports Metal, otherwise it should be skipped.

Reply:
Yeah, even if the *machine* supports metal, we should skip if the target doesn't support it, eg 

```
    if (!get_jit_target_from_environment().has_feature(Target::Metal)) {
        printf("[SKIP] error/metal_threads_too_large ignored for targets without Metal enabled.\n");
        _halide_user_assert(0);
    }
```

Reply:
Thank you, since now it only checks in debug runtime, so I add a specific target, like "metal_vector_too_large" did.